torch/nn/parallel/distributed.py

```diff
@@ -406,9 +408,9 @@ def __init__(
             self.device_ids = None
             self.output_device = None
         else:
-            # Use all devices by default for single-device GPU modules
+            # Use the current device by default for single-device GPU modules
             if device_ids is None:
-                device_ids = _get_all_device_indices()
+                device_ids = [_get_current_device_index()]
```

Comment:
We should document this behavior in our docs. Basically if device_ids is None and it is a single device module, it will use the current GPU device as the device for the model, so you need to use `torch.cuda.set_device` to set the appropriate device before you call DDP.

Another option would be to expect the user to place the model on the appropriate device for single device models (similar to how we expect this for multi-device models today). cc @mrshenli Thoughts?

Reply:
> We should document this behavior in our docs. Basically if device_ids is None and it is a single device module, it will use the current GPU device as the device for the model, so you need to use `torch.cuda.set_device` to set the appropriate device before you call DDP.

Thanks for the suggestion! Added this requirement to the doctoring of DDP (in `device_ids` arg section).

> Another option would be to expect the user to place the model on the appropriate device for single device models (similar to how we expect this for multi-device models today).

In the pitch, it says "If device_ids is not specified, DDP will use the provided model (module argument in DDP constructor) as-is, regardless if the model is on one GPU or multiple GPUs or on CPU." Therefore, I think `None` should be allowed for convenience.

Reply:
vote for assuming user to place the model and input on the appropriate device for single device model when device_ids is none. This is consistent with multi-device models. also possibly better than assuming user has called torch.cuda.set_device()